### PR TITLE
Integrations specs and fix the n+1 problem

### PR DESCRIPTION
**Pull request by dieum512**
In this milestone we:

- [x] Wrote integration tests with Capybara gem,
- [x] Secured the app from n+1 problems and
- [x] Fixed some linter errors

Thank you for taking the time to review this project :wink: 

### Here is the comment  before fixing the n+1 problem
Screenshot from 2023-08-28 08-45-02

### Here is the comment  after fixing the n+1 problem
Screenshot from 2023-08-28 08-49-02

### Here is the post  before fixing the n+1 problem
Screenshot from 2023-08-28 08-51-48

### Here is the post  after fixing the n+1 problem
Screenshot from 2023-08-28 08-53-27

### In conclusion, I add this screenshot to show all the tests passing
Screenshot from 2023-08-29 10-21-48

### CODE REVIEW

**Review by sumairq — APPROVED** [review]
### Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Great job
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
